### PR TITLE
Changes for digital subscription gifting

### PR DESCRIPTION
**Pull request by i-hardy**
## What does this change?
This adds the necessary changes to the account overview page to support digital subscription gifting. For the gifter, we are showing the date of purchase but not start/end dates, and removing links to payment management as it's a one-time payment. For the giftee, we are showing the start and end dates relative to when they redeem the subscription, and a payment method of 'gift redemption'. The cards show the same 'Gift' flag and icon as for Guardian Weekly gifting.

## How to test
Gifted digital subscriptions can be purchased on CODE at https://support.code.dev-theguardian.com/uk/subscribe/digital/gift.

### Gifter view

**Mobile**
Screenshot 2020-11-09 at 10 32 17

**Tablet**
Screenshot 2020-11-09 at 10 24 41

**Desktop**
Screenshot_2020-11-09 My Account The Guardian

### Giftee view

**Mobile**
Screenshot 2020-11-09 at 10 33 36

**Tablet**
Screenshot 2020-11-09 at 10 43 39

**Desktop**
Screenshot_2020-11-09 My Account The Guardian(1)

### CODE REVIEW

File: app/client/components/accountoverview/accountOverviewCard.tsx [hunk 1]
```diff
@@ -51,13 +51,26 @@ export const AccountOverviewCard = (props: AccountOverviewCardProps) => {
...
+  // We want to hide manage sub/payment buttons and the trial period if the product is a gifted digi sub
+  const isDigiSubGift =
+    specificProductType.friendlyName === "digital subscription" &&
+    isGift(props.productDetail.subscription);
+
+  const userIsGifter = isDigiSubGift && props.productDetail.isPaidTier;
+
+  const giftPurchaseDate = props.productDetail.subscription.lastPaymentDate;
+
+  const shouldNotShowStartDate = shouldShowJoinDateNotStartDate || userIsGifter;
```
Comment (rupertbates) [inline app/client/components/accountoverview/accountOverviewCard.tsx:72]: As we negate this when we use it perhaps it should just be `shouldShowStartDate`

**Review by rupertbates — APPROVED** [review]
Just one naming suggestion but otherwise 👍

**Comment by twrichards** [comment]
Looks lovely, but ideally the account overview component should remain as generic as possible, delegating 'per product' differences to the ProductTypes definitions (see https://github.com/guardian/manage-frontend/wiki/Product-Types).

Where you need variables from the account overview, consider creating function properties in ProductTypes. Happy to pair to explain further...

**Comment by i-hardy** [comment]
Thanks @twrichards- turns out we don't actually want to be discriminating on DS vs GW gifting, so I've taken out the product-specific code. Is there anything else you think needs to change or am I good to go ahead and merge?

File: app/client/components/accountoverview/accountOverviewCard.tsx [hunk 2]
```diff
@@ -406,12 +431,19 @@ export const AccountOverviewCard = (props: AccountOverviewCardProps) => {
...
+              <ul css={keyValuePairCss}>
+                <li css={keyCss}>Payment</li>
+                <li css={valueCss}>Gift redemption</li>
+              </ul>
+            ) : (
+              <ul css={keyValuePairCss}>
+                <li css={keyCss}>Payment</li>
+                <li css={valueCss}>FREE</li>
+              </ul>
+            ))}
```
Comment (twrichards) [inline app/client/components/accountoverview/accountOverviewCard.tsx:446]: This avoids some repetition...
```suggestion
          {!props.productDetail.isPaidTier && (
              <ul css={keyValuePairCss}>
                <li css={keyCss}>Payment</li>
                <li css={valueCss}>{isGifted ? "Gift redemption" : "FREE"}</li>
              </ul>
            )}
```

File: app/client/components/accountoverview/accountOverviewCard.tsx [hunk 3]
```diff
@@ -373,7 +398,7 @@ export const AccountOverviewCard = (props: AccountOverviewCardProps) => {
                   )}
                 </li>
               </ul>
-              {!props.productDetail.subscription.payPalEmail && (
+              {!props.productDetail.subscription.payPalEmail && !isGifted && (
```
Comment (twrichards) [inline app/client/components/accountoverview/accountOverviewCard.tsx:401]: does suggest that some gifted subscriptions come through with `props.productDetail.isPaidTier` = `true` ??

Comment (twrichards) [reply app/client/components/accountoverview/accountOverviewCard.tsx:401]: or is that the difference between the buyer seeing the gift and the recipient having 'redeemed' the gift?

Comment (twrichards) [reply app/client/components/accountoverview/accountOverviewCard.tsx:401]: I guess line 65 answers my question 👍

File: app/client/components/accountoverview/accountOverviewCard.tsx [hunk 4]
```diff
@@ -273,27 +296,29 @@ export const AccountOverviewCard = (props: AccountOverviewCardProps) => {
...
-              onClick={() =>
-                trackEvent({
-                  eventCategory: "account_overview",
-                  eventAction: "click",
-                  eventLabel: `manage_${groupedProductType.urlPart}`
-                })
-              }
-            />
-          </div>
+          {!isGifted && (
```
Comment (twrichards) [inline app/client/components/accountoverview/accountOverviewCard.tsx:299]: are we sure we don't want gift subs to be individually manageable? (from there one can get to features like holiday stops etc.)

I'm conscious this is changing existing behaviour [I think] (perhaps that existing behaviour wasn't desirable, but it's probably worth checking with @rBangay and Jian)

**Review by twrichards — APPROVED** [review]
Got a few comments, no blockers per-se (my previous concern around per-product stuff in `accountOverviewCard.tsx` has been resolved 🙌 ).

As stated before, excellent to see cross-device screenshots 🏅

File: app/client/components/accountoverview/accountOverviewCard.tsx (hunk 4 above)
Comment (i-hardy) [reply app/client/components/accountoverview/accountOverviewCard.tsx:299]: That's the requirement from Gus and Matt. All gift subscriptions, whether digital or Guardian Weekly, are one-off payments, so there's nothing really to manage- if you want to buy another one you go through the checkout again.

Comment (twrichards) [reply app/client/components/accountoverview/accountOverviewCard.tsx:299]: fair enough payment shouldn't be updatable - but i suspect holiday stops and reporting delivery problems might be desired (although not sure they're supported for gifts right now - @arausuy @rBangay )?

**Review by twrichards — APPROVED** [review]

**Pull request merged**